### ISSUE TITLE
`isLibrary` and `esmodule` still emits runtime for Worker and async

### ISSUE THREAD

**Issue by mischnic** [parcel-bundler/parcel#4349]
# 🐛 bug report

`isLibrary` still emits helpers to get the URL at which async dependencies should be requested in the browser.

(cc @wbinnssmith)

## 🤔 Expected Behavior

Output is bundler-compatible (`new Worker(new URL("string literal", import.meta.url)`, `import("./string-literal")`)

## 😯 Current Behavior

We don't want `getBundleURL`, `getWorkerUrl` in this configuration.

## 💻 Code Sample

```js
// index.js
new Worker("./worker.js");
import("./async.js");

// async.js & worker.js: anything
```
```json5
// package.json
{
	"app": "dist/index.js",
	"targets": {
		"app": {
			"isLibrary": true,
			"outputFormat": "esmodule"
		}
	}
}
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 9007d6ac51a44cca44ff216ee103baca76be5504

**Comment by Banou26** [comment]
If we could add this example in the documentations for https://v2.parceljs.org/getting-started/library/ this would be really nice, I've been searching on how to export ESM for a while but never actually found out how until now.
I was also wondering if exporting TS types could be in the scope of Parcel or kept to TSC?

**Comment by mischnic** [comment]
> If we could add this example in the documentations for v2.parceljs.org/getting-started/library

That is indeed the plan for that page.

> I was also wondering if exporting TS types could be in the scope of Parcel or kept to TSC?

You mean like https://v2.parceljs.org/languages/typescript#generating-typings ?

For example https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/button/package.json

**Comment by Banou26** [comment]
Oh, thanks for the example.
I've tried using this with a target but i haven't found a way to make it work.
Replaced my target with `main` and it worked like a charm, thanks!

**Comment by devongovett** [comment]
Should be fixed by #6531.
